sklearn/cluster/_dbscan_inner.pyx

```diff
@@ -38,12 +40,15 @@ def dbscan_inner(np.ndarray[np.uint8_t, ndim=1, mode='c'] is_core,
                     neighb = neighborhoods[i]
                     for i in range(neighb.shape[0]):
                         v = neighb[i]
-                        if labels[v] == -1:
+                        if labels[v] == -1 and seen.count(v) == 0:
+                            seen.insert(v)
```

Comment:
if you say so...

btw I just noticed that the p parameter in not documented in the docstring

can you fix it